packages/flutter/lib/src/animation/curves.dart

```diff
@@ -305,6 +305,97 @@ class Cubic extends Curve {
   }
 }
 
+/// A cubic polynomial composed of two curves that share a common center point,
+/// mapping of the unit interval.
+///
+/// The curve runs through three points: (0,0), the [midpoint], and (1,1).
+///
+/// The [Curves] class contains a curve defined with this class:
+///
+///  * [Curves.easeInOutCubicEmphasized]
+///
+/// The [ThreePointCubic] class implements third-order Bézier curves, where two
+/// curves share an interior [midpoint] that the curve passes through. If the
+/// control points surrounding the middle point ([b1], and [a2]) are not
+/// colinear with the middle point, then the curve's derivative will have a
+/// discontinuity (a cusp) at the shared middle point.
+///
+/// See also:
+///
+///  * [Curves], where many more predefined curves are available.
+///  * [Cubic], which defines a single cubic polynomial.
+///  * [CatmullRomCurve], a curve which passes through specific values.
+class ThreePointCubic extends Curve {
+  /// Creates two cubic curves that share a common control point.
+  ///
+  /// Rather than creating a new instance, consider using one of the common
+  /// three-point cubic curves in [Curves].
+  ///
+  /// The arguments correspond to the control points for the two curves,
+  /// including the [midpoint], but do not include the two implied end points at
+  /// (0,0) and (1,1), which are fixed.
+  const ThreePointCubic(this.a1, this.b1, this.midpoint, this.a2, this.b2);
+
+  /// The coordinates of the first control point of the first curve.
+  ///
+  /// The line through the point (0, 0) and this control point is tangent to the
+  /// curve at the point (0, 0).
+  final Offset a1;
+
+  /// The coordinates of the second control point of the first curve.
+  ///
+  /// The line through the [midpoint] and this control point is tangent to the
+  /// curve approaching the [midpoint].
+  final Offset b1;
+
+  /// The coordinates of the middle shared point.
+  ///
+  /// The curve will go through this point. If the control points surrounding
+  /// this middle point ([b1], and [a2]) are not colinear with this point, then
+  /// the curve's derivative will have a discontinuity (a cusp) at this point.
+  final Offset midpoint;
+
+  /// The coordinates of the first control point of the second curve.
+  ///
+  /// The line through the [midpoint] and this control point is tangent to the
+  /// curve approaching the [midpoint].
+  final Offset a2;
+
+  /// The coordinates of the second control point of the second curve.
+  ///
+  /// The line through the point (1, 1) and this control point is tangent to the
+  /// curve at (1, 1).
+  final Offset b2;
+
+  @override
+  double transformInternal(double t) {
+    final bool firstCurve = t < midpoint.dx;
```

Comment:
Should t be the same as x?

Reply:
No, `t` is the parametric value we're evaluating at, and it is the traditional variable name for a parametric value. It's also what is used in the base class declaration, so I'd get a warning if I didn't use it.

I often wish we had some more "point" classes, so that I could use a `Parametric` or something that had members called `s` and `t`, or a `Point` with `x` and `y`. These are only "offsets" because that's the only point class we have.

Reply:
Maybe I should use a `Point<double>` for this (from `dart:math`)?

Reply:
That looks pretty ugly, though:
```dart
  static const ThreePointCubic easeInOutCubicEmphasized = ThreePointCubic(
    math.Point<double>(0.05, 0), math.Point<double>(0.133333, 0.06),
    math.Point<double>(0.166666, 0.4),
    math.Point<double>(0.208333, 0.82), math.Point<double>(0.25, 1),
  );
```

Reply:
I see now: the X axis is t, and the Y axis is S(t) - the value of the parametric curve. Thanks for the explanation!